feature/home/src/main/java/io/filmtime/feature/home/HomeScreen.kt

```diff
@@ -34,11 +50,46 @@ fun HomeScreen(
   onMovieClick: (tmdbId: Int) -> Unit,
   onShowClick: (tmdbId: Int) -> Unit,
 ) {
+
+
   val state by viewModel.state.collectAsStateWithLifecycle()
+  var load by remember { mutableStateOf(false) }
+
+
+  LaunchedEffect(load) {
+    if (load) {
+      viewModel.retry()
+      viewModel.loadTrendingMovies()
+      viewModel.loadTrendingShows()
+      load = false
+
+    }
+  }
 
+  val context = LocalContext.current
   if (state.isLoading) {
+
     LoadingVideoSectionRow(numberOfSections = 2)
-  } else {
+  }else if (state.error != null){
+    when( state.error){
+      is GeneralError.NetworkError->{
+        NetworkErroScreen(context){
+          load = true
+        }
+
+      }
+      is GeneralError.ApiError -> {
+        ApiErrorScreen{
+          load= true
+        }
+      }
+     is GeneralError.UnknownError ->{
+       UnknownErrorScreen()
+     }
+
+      else -> {}
```

Comment:
It's not advisable to include an else statement here. If we were to add another state to GeneralError, it could easily be overlooked, leading to potential issues.

Reply:
i have removed else branch but  android studio suggests to add "null -> TODO()" or else branch  so i added null branch otherwise it does not compile